### PR TITLE
TSQL parsing + symbol for concatenation or arithmetic operation

### PR DESCRIPTION
**Pull request by 10bas10**
Hi All,

Im facing quite a challenge for parsing the + symbol in T-SQL. I would love to get some help on my approach.

The problem is that there are two kind of contexts possible when using the + symbol:
- For arithmetic operations such as column1 + 234
- For string concatenations such as column1 + '234'

The problem is to identify it is either a + symbol for concatenation or not. In case of a string concatenation I managed to translate this to DPipe and is handled ok. However you would need to scan for string values in order to determine the context of string concatenation. This will never be a 100% solution as you would need the datatypes of the columns involved. However im trying a best effort approach here.

My approach is to determine the related set of literals and determine whether its a string concatenation based on that. After I would like to replace all of the exp.Add with a exp.DPipe. 

Do you guys have any good ideas on this problem?

Thanks!

### CODE REVIEW

**Comment by tobymao** [comment]
this cannot be done this way. this can only be done with type inference

**Comment by tobymao** [comment]
for the simple case, it’s better to do this as a transform as opposed to at parse time

File: sqlglot/dialects/tsql.py [hunk 1]
```diff
@@ -219,6 +220,50 @@ def _parse_convert(self, strict):
             # Entails a simple cast without any format requirement
             return self.expression(exp.Cast if strict else exp.TryCast, this=this, to=to)
 
+        def _parse_term(self):
```
Comment (tobymao) [inline sqlglot/dialects/tsql.py:223]: take a look at sqlglot/transforms and how that is used. something like this would go there

Comment (10bas10) [reply sqlglot/dialects/tsql.py:223]: Ill have a look thanks!

**Pull request closed without merging**